hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-router/src/main/java/org/apache/hadoop/yarn/server/router/ssproxy/RouterStateStoreProxyWebServices.java

```diff
@@ -0,0 +1,412 @@
+import org.apache.hadoop.yarn.server.federation.store.records.dao.SubClusterDeregisterDAO;
+import org.apache.hadoop.yarn.server.federation.store.records.dao.SubClusterHeartbeatDAO;
+import org.apache.hadoop.yarn.server.federation.store.records.dao.SubClusterInfoDAO;
+import org.apache.hadoop.yarn.server.federation.store.records.dao.SubClusterPolicyConfigurationDAO;
+import org.apache.hadoop.yarn.server.router.webapp.RouterWebServices;
+import org.apache.hadoop.yarn.util.LRUCacheHashMap;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+
+/**
+ * RouterStateStoreProxyWebServices is a service that runs on each router that
+ * can be used to intercept and inspect {@link StateStoreWebServiceProtocol}
+ * messages from client to the cluster FederationStateStore. It listens
+ * {@link StateStoreWebServiceProtocol} REST messages from the client and
+ * creates a request intercepting pipeline instance for each client. The
+ * pipeline is a chain of {@link StateStoreProxyInterceptor} instances that can
+ * inspect and modify the request/response as needed. The main difference with
+ * AMRMProxyService is the protocol they implement.
+ */
+@Singleton
+@Path(HttpProxyFederationStateStoreConsts.ROOT)
+public class RouterStateStoreProxyWebServices
+    implements StateStoreWebServiceProtocol {
+  private static final Logger LOG =
+      LoggerFactory.getLogger(RouterStateStoreProxyWebServices.class);
+  private final Router router;
+  private final Configuration conf;
+  private @Context HttpServletResponse response;
+
+  private Map<String, RequestInterceptorChainWrapper> userPipelineMap;
+
+  @Inject
+  public RouterStateStoreProxyWebServices(final Router router,
+      Configuration conf) {
+    this.router = router;
+    this.conf = conf;
+    int maxCacheSize =
+        conf.getInt(YarnConfiguration.ROUTER_PIPELINE_CACHE_MAX_SIZE,
+            YarnConfiguration.DEFAULT_ROUTER_PIPELINE_CACHE_MAX_SIZE);
+    this.userPipelineMap = Collections.synchronizedMap(
+        new LRUCacheHashMap<String, RequestInterceptorChainWrapper>(
+            maxCacheSize, true));
+  }
+
+  /**
+   * Returns the comma separated intercepter class names from the configuration.
+   *
+   * @param config
+   * @return the intercepter class names as an instance of ArrayList
+   */
+  private List<String> getInterceptorClassNames(Configuration config) {
+    String configuredInterceptorClassNames = config
+        .get(YarnConfiguration.ROUTER_SSPROXY_INTERCEPTOR_CLASS_PIPELINE,
+            YarnConfiguration.DEFAULT_ROUTER_SSPROXY_INTERCEPTOR_CLASS);
+
+    List<String> interceptorClassNames = new ArrayList<String>();
+    Collection<String> tempList =
+        StringUtils.getStringCollection(configuredInterceptorClassNames);
+    for (String item : tempList) {
+      interceptorClassNames.add(item.trim());
```

Comment:
Extract item.trim() into somehting like: interceptorClassName

Reply:
BTW, why can't we just use the the collection coming from getStringCollection()?
Configuration is also able to return a list of strings from Configuration#getStringCollection().